### PR TITLE
Pressure plates can be placed on fences and hoppers

### PR DESCRIPTION
**Pull request by Bond-009**
This fixes issue #2469

### CODE REVIEW

File: src/Blocks/BlockPressurePlate.h [hunk 1]
```diff
@@ -28,7 +28,26 @@ class cBlockPressurePlateHandler :
 			return false;
 		}
 
-		return (cBlockInfo::FullyOccupiesVoxel(a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ)));
+		// TODO: check if the block is upside-down slab or upside-down stairs
+		BLOCKTYPE Block = a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ);
```
Comment (madmaxoft) [inline src/Blocks/BlockPressurePlate.h]: Thies needs a check (or a change to the check above) - if `a_RelY` is zero, this will crash.

Comment (Bond-009) [inline src/Blocks/BlockPressurePlate.h]: Do you mean `a_RelY - 1`?

Comment (madmaxoft) [inline src/Blocks/BlockPressurePlate.h]: `a_RelY - 1` must be at least zero for this to work (Y coord ranges from 0 to 255). So when attempting to place a pressure plate at the bottom-most layer of the world, your code will read a block at Y coord `-1`; that is definitely not good and can crash the server.

Comment (Bond-009) [inline src/Blocks/BlockPressurePlate.h]: It should be fixed now.

**Pull request merged**